GREYErrorFormatter.h

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "GREYError.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface GREYErrorFormatter: NSObject
+
+/**
+ * For a given @c hierarchy, create a human-readable hierarchy
+ * including the legend.
+ *
+ * @param hierarchy The string representation of the App UI Hierarchy
+ *
+ * @return The formatted hierarchy and legend to be printed on the console
+*/
++ (NSString *)formattedHierarchy:(nonnull NSString *)hierarchy;
```

Comment:
Prefer C API to class method. Class method created redundant symbols/encoding/runtime, so let's avoid it if it's not necessary. Extended reading:
https://medium.com/@dmaclach/objective-c-encoding-and-you-866624cc02de